op-alt-da/damock.go

```diff
@@ -99,3 +103,84 @@ func (d *AltDADisabled) OnFinalizedHeadSignal(f HeadSignalFn) {
 func (d *AltDADisabled) AdvanceL1Origin(ctx context.Context, l1 L1Fetcher, blockId eth.BlockID) error {
 	return ErrNotEnabled
 }
+
+// FakeDAServer is a fake DA server for e2e tests.
+// It is a small wrapper around DAServer that allows for setting request latencies,
+// to mimic a DA service with slow responses (eg. eigenDA with 10 min batching interval).
+type FakeDAServer struct {
+	*DAServer
+	putRequestLatency time.Duration
+	getRequestLatency time.Duration
+}
+
+func NewFakeDAServer(host string, port int, log log.Logger) *FakeDAServer {
+	store := NewMemStore()
+	fakeDAServer := &FakeDAServer{
+		DAServer:          NewDAServer(host, port, store, log, true),
+		putRequestLatency: 0,
+		getRequestLatency: 0,
+	}
+	return fakeDAServer
+}
+
+func (s *FakeDAServer) HandleGet(w http.ResponseWriter, r *http.Request) {
+	time.Sleep(s.getRequestLatency)
```

Comment:
Isn't this going to cause tests to take 5 seconds times the number of times this is called extra amount of time to run?

Reply:
I understand that because of the newly introduced concurrency in sending DA transactions, multiple submit routines would wait those 5 secs in parallel.

Reply:
Correct, this is only used in the concurrent da requests test.